crates/uv-resolver/src/lock/mod.rs

```diff
@@ -965,13 +968,41 @@ impl Lock {
 
         let mut packages = ArrayOfTables::new();
         for dist in &self.packages {
+            let dist = match proxies.as_ref().and_then(|proxies| {
+                proxies.iter().find(|p| {
+                    dist.registry_source_index_url()
+                        .is_ok_and(|url| url.is_some_and(|url| url == p.url))
+                })
+            }) {
+                Some(proxy) => &dist.with_canonical_urls(proxy).expect("FIXME"),
```

Comment:
If we had a method like `with_canonical_urls` that took `mut self` on the `Lock`, could we avoid having to re-clone the entire distribution here?

Reply:
I've changed this and related methods to mutate directly without cloning